Public/Src/Sandbox/Windows/DetoursServices/DetouredFunctions.cpp

```diff
@@ -1778,6 +1778,224 @@ NTSTATUS NTAPI Detoured_ZwSetInformationFile(
+                trim_inplace(command);
+                commandArgs = wstring();
+            }
+            else
+            {
+                wstring noQuoteCommand = fullCommandLine.substr(1, closeQuoteIndex - 1);
+
+                // Find the next delimiting space after the close double-quote.
+                // For example a command line like "c:\program files"\foo we need to
+                // keep \foo and cut the quotes to produce c:\program files\foo
+                size_t spaceDelimiterIndex = fullCommandLine.find(L' ', closeQuoteIndex + 1);
+                if (spaceDelimiterIndex == wstring::npos)
+                {
+                    // No space, take everything through the end of the command line.
+                    spaceDelimiterIndex = fullCommandLine.length();
+                }
+
+                command = (noQuoteCommand +
+                    fullCommandLine.substr(closeQuoteIndex + 1, spaceDelimiterIndex - closeQuoteIndex));
+                trim_inplace(command);
+                commandArgs = fullCommandLine.substr(spaceDelimiterIndex + 1);
+                trim_inplace(commandArgs);
+            }
+        }
+        else
+        {
+            // No close quote. Take everything through the end of the command line as the command.
+            command = fullCommandLine.substr(1);
+            commandArgs = wstring();
+        }
+    }
+    else
+    {
+        // No open quote, pure space delimiter.
+        size_t spaceDelimiterIndex = fullCommandLine.find(' ');
+        if (spaceDelimiterIndex == wstring::npos)
+        {
+            // No space, take everything through the end of the command line.
+            spaceDelimiterIndex = fullCommandLine.length();
+        }
+
+        command = fullCommandLine.substr(0, spaceDelimiterIndex);
+        commandArgs = fullCommandLine.substr(spaceDelimiterIndex + 1);
+        trim_inplace(commandArgs);
+    }
+}
+
+static bool CommandArgsContainMatch(const wchar_t *commandArgs, const wchar_t *argMatch)
+{
+    if (argMatch == nullptr)
+    {
+        // No optional match, meaning always match.
+        return true;
+    }
+
+    return wcsstr(commandArgs, argMatch) != nullptr;
+}
+
+static bool ShouldShim(const wstring &command, const wchar_t *commandArgs)
+{
+    // Easy cases.
+    if (g_pShimProcessMatches == nullptr || g_pShimProcessMatches->empty())
+    {
+        // Shim everything or shim nothing if there are no matches to compare.
+        return g_ProcessExecutionShimAllProcesses;
+    }
+
+    size_t commandLen = command.length();
+
+    bool foundMatch = false;
+
+    for (auto it = g_pShimProcessMatches->begin(); it != g_pShimProcessMatches->end(); ++it)
+    {
+        ShimProcessMatch *pMatch = *it;
+
+        const wchar_t *processName = pMatch->ProcessName.get();
+        size_t processLen = wcslen(processName);
+
+        // lpAppName is longer than e.g. "cmd.exe", see if lpAppName ends with e.g. "\cmd.exe"

```

Comment:
>\cmd.exe [](start = 85, length = 8)

Just checking if the PATHEXT logic happens in the shell or if it also happens at this layer. if so you probably need to do more matching as `cmd` would look in order or `pathext` environment for existing files.

Reply:
The shim I have (not visible in this PR) actually has to run the PATHEXT logic itself since it's not using ShellExecute.